tests/unit_tests/db_engine_specs/test_kusto.py

```diff
@@ -0,0 +1,104 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+# pylint: disable=unused-argument, import-outside-toplevel, protected-access
+import pytest
+from flask.ctx import AppContext
+
+
+@pytest.mark.parametrize(
+    "sql,expected",
+    [
+        ("SELECT foo FROM tbl", True),
+        ("SHOW TABLES", False),
+        ("EXPLAIN SELECT foo FROM tbl", False),
+        ("INSERT INTO tbl (foo) VALUES (1)", False),
+    ],
+)
+def test_sql_is_readonly_query(
```

Comment:
Could we add a few more tests for the SQL spec, like a few timegrains, `convert_dttm` etc, just to get coverage up? You can probably copy some template tests from the pre-existing unit tests.